vpr/src/route/lookahead_profiler.cpp

```diff
@@ -0,0 +1,173 @@
+        int delta_y = to_y - from_y;
+
+        // Calculate the actual cost, delay, and congestion from the current node to the sink.
+        float actual_cost = total_backward_cost - current_backward_cost;
+        float actual_delay = total_backward_delay - current_backward_delay;
+        float actual_congestion = total_backward_congestion - current_backward_congestion;
+
+        // Get the cost, delay, and congestion estimates made by the lookahead.
+        // Note: lookahead_cost = lookahead_delay * criticality + lookahead_congestion * (1. - criticality)
+        float lookahead_cost = router_lookahead.get_expected_cost(curr_inode, sink_inode, cost_params, 0.0);
+        auto [lookahead_delay, lookahead_congestion] = router_lookahead.get_expected_delay_and_cong(curr_inode, sink_inode, cost_params, 0.0);
+
+        // Get the current node's type and length
+        std::string node_type_str = rr_graph.node_type_string(curr_inode);
+        std::string node_length = (node_type_str == "CHANX" || node_type_str == "CHANX")
+                                      ? std::to_string(rr_graph.node_length(curr_inode))
+                                      : "--";
+
+        /* Write out all info */
+
+        lookahead_verifier_csv << iteration << ",";            // iteration no.
+        lookahead_verifier_csv << source_inode << ",";         // source node
+        lookahead_verifier_csv << sink_inode << ",";           // sink node
+        lookahead_verifier_csv << block_name << ",";           // sink block name
+        lookahead_verifier_csv << atom_block_model << ",";     // sink atom block model
+        lookahead_verifier_csv << cluster_block_type << ",";   // sink cluster block type
+        lookahead_verifier_csv << tile_height << ",";          // sink cluster tile height
+        lookahead_verifier_csv << tile_width << ",";           // sink cluster tile width
+        lookahead_verifier_csv << curr_inode << ",";           // current node
+        lookahead_verifier_csv << node_type_str << ",";        // node type
+        lookahead_verifier_csv << node_length << ",";          // node length
+        lookahead_verifier_csv << i << ",";                    // num. nodes from sink
+        lookahead_verifier_csv << delta_x << ",";              // delta x
+        lookahead_verifier_csv << delta_y << ",";              // delta y
+        lookahead_verifier_csv << actual_cost << ",";          // actual cost
+        lookahead_verifier_csv << actual_delay << ",";         // actual delay
+        lookahead_verifier_csv << actual_congestion << ",";    // actual congestion
+        lookahead_verifier_csv << lookahead_cost << ",";       // predicted cost
+        lookahead_verifier_csv << lookahead_delay << ",";      // predicted delay
+        lookahead_verifier_csv << lookahead_congestion << ","; // predicted congestion
+        lookahead_verifier_csv << cost_params.criticality;     // criticality
+        lookahead_verifier_csv << std::endl;
```

Comment:
std::endl flushes the buffer
avoid using it if it isn't necessary to flush the buffer